mlflow/tensorflow/__init__.py

```diff
@@ -138,6 +140,29 @@ def log_model(
     """
     Log a TF2 core model (inheriting tf.Module) or a Keras model in MLflow Model format.
 
+    .. note::
+
+        If you log a Keras or TensorFlow model without a signature, inference with
+        :py:func:`mlflow.pyfunc.spark_udf()` will not work unless the model's pyfunc
+        representation accepts pandas DataFrames as inference inputs.
+
+        You can infer a model's signature by calling the :py:func:`mlflow.models.infer_signature()`
+        API on features from the model's test dataset. You can also manually create a model
+        signature, for example:
+
+        .. code-block:: python
+            :caption: Example of creating signature for saving TensorFlow and `tf.Keras` models
+
+            from mlflow.types.schema import Schema, TensorSpec
+            from mlflow.models.signature import ModelSignature
+            input_schema = Schema(
+                [
+                    TensorSpec(np.dtype(np.uint64), (-1, 5), "field1"),
+                    TensorSpec(np.dtype(np.float32), (-1, 3, 2), "field2"),
+                ]
+            )
+            signature = ModelSignature(inputs=input_schema)
```

Comment:
Just showing code to create a signature is not helpful because it's unclear why `(-1, 5)` and `(-1, 3, 2)` are chosen.

Reply:
I'd add code for creating a model that has this signature

Reply:
We can create an example code for this. Otherwise the doc block here will be too long.

Reply:
Can you create an example script?

Reply:
or add comments in the code.

Reply:
ok I added comments.